Source/BidMachineAdapterInterstitialAd.swift

```diff
@@ -0,0 +1,131 @@
+// Copyright 2023-2023 Chartboost, Inc.
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file.
+
+import ChartboostMediationSDK
+import Foundation
+import BidMachine
+
+final class BidMachineAdapterInterstitialAd: BidMachineAdapterAd, PartnerAd {
+    
+    /// The BidMachineSDK ad instance.
+    private var ad: BidMachineInterstitial?
+    
+    /// Loads an ad.
+    /// - parameter viewController: The view controller on which the ad will be presented on. Needed on load for some banners.
+    /// - parameter completion: Closure to be performed once the ad has been loaded.
+    func load(with viewController: UIViewController?, completion: @escaping (Result<PartnerEventDetails, Error>) -> Void) {
+        log(.loadStarted)
+
+        // Make request configuration
+        let config: BidMachineRequestConfigurationProtocol
+        do {
+            config = try BidMachineSdk.shared.requestConfiguration(.interstitial)
+        } catch {
+            let chartboostMediationError = self.error(.loadFailureUnknown, error: error)
+            self.log(.loadFailed(chartboostMediationError))
+            completion(.failure(chartboostMediationError))
```

Comment:
nit:
Inside a `catch` statement error should be non-optional, so we could pass the partner error directly.
```suggestion
            log(.loadFailed(error))
            completion(.failure(error))
```